### PR TITLE
Breaking change: config remote support more clients

### PR DESCRIPTION
**Pull request by GGXXLL**
I'm not sure if we need so many clients, but I've implemented it, so give some suggestions.

Reference [viper.remote](https://github.com/spf13/viper/blob/master/remote/remote.go)

Support client:
 - [etcd](https://pkg.go.dev/go.etcd.io/etcd/client/v3)
 - [redis](https://pkg.go.dev/github.com/go-redis/redis/v8)
 - [consul](https://pkg.go.dev/github.com/hashicorp/consul/api)
 - [firestore](https://pkg.go.dev/cloud.google.com/go/firestore)

### CODE REVIEW

File: config/remote/example_test.go [hunk 1]
```diff
@@ -20,13 +21,17 @@ func Example() {
...
+		DialTimeout: 5 * time.Second,
+	}, key, "name: etcd")
 
-	c := core.New(etcd.WithKey(cfg, key, yaml.Codec{}))
+	cfg := remote.Config{
+		Endpoints: envEtcdAddrs,
+		Name:      "etcd",
+		Key:       key,
+	}
+	c := core.New(remote.WithKey(cfg, yaml.Codec{}))
```
Comment (Reasno) [inline config/remote/example_test.go:34]: why change the signature of WithKey?

**Comment by Reasno** [comment]
Viper is not a good candidate for example. It has so many bad designs that cannot be changed due to API compatibility.

**Review by Reasno — COMMENTED** [review]
forking is not maintaining. crypt is a big package and may easily contain hidden vulnerabilities. I'd rather not include anything from github.com/DoNewsCode/crype.

File: config/remote/example_test.go (hunk 1 above)
Comment (GGXXLL) [reply config/remote/example_test.go:34]: Thank you for your review.
- I agree with you on `viper`. The original `crypt` is not very good. I have rebuilt the code of his `backend` part, and added related tests. Maybe I should write the `backend` part in the core?
- Add `Name`( for choice backend client ) and `WatchInterval` ( for timer selector of Watch). A little too many parameters. So I packed them into `remote.Config`.

**Pull request closed without merging**